shared/src/main/scala/amf/client/remod/AMFClient.scala

```diff
@@ -11,29 +11,29 @@ import amf.core.resolution.pipelines.ResolutionPipeline
 import amf.core.services.RuntimeCompiler
 import amf.core.validation.AMFValidationReport
 
-import scala.concurrent.Future
-
+import scala.concurrent.{ExecutionContext, Future}
 
 // client
-private[remod] class AMFClient(env: BaseEnvironment) {
+private[amf] class AMFClient(protected val configuration: AMFConfiguration) {
 
+  implicit val exec: ExecutionContext = configuration.getExecutionContext
 
-  def getEnvironment:BaseEnvironment = env
+  def getEnvironment: AMFConfiguration = configuration
   // how do we return the ErrorHandler that is created by the provider?
   // relation between EH vs BU? should the BU have an EH inside?
   // should parse always return error handler + base unit? (Amf result)
 
   // sync or async?
 
   // content type format, pendiente
-  def parse(url: String, vendor: Option[Vendor] = None): Future[AmfResult] = AMFParser.parse(url, env)
-    // build parsing context?
+  def parse(url: String, vendor: Option[Vendor] = None): Future[AMFResult] = AMFParser.parse(url, configuration)
```

Comment:
vendor variable is used? The called AMFParser.parse() method didn't receive it, but there is one